File: Jan/Week 3/Extra/problem_1.py
# Day 15 Problem No 32

# 1431. Kids With the Greatest Number of Candies => https://leetcode.com/problems/kids-with-the-greatest-number-of-candies/description/?envType=study-plan-v2&envId=leetcode-75

class Solution(object):
    def kidsWithCandies(self, candies, extraCandies):
        """
        :type candies: List[int]
        :type extraCandies: int
        :rtype: List[bool]
        """
        maxC = 0;
        for i in candies:
            if i > maxC:
                maxC = i
        ans = []
        for i in candies:
            if i+extraCandies >= maxC:
                ans.append(True)
            else:
                ans.append(False)
        return ans
        

File: Jan/Week 3/LeetCode/problem_1.cpp
//  Day 15 Problem No 30

// 605. Can Place Flowers => https://leetcode.com/problems/can-place-flowers/description/?envType=study-plan-v2&envId=leetcode-75

class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        for(int i=0;i<flowerbed.size();i++){
            if(flowerbed[i] == 0 && (i==0 || flowerbed[i-1] == 0) && (i == flowerbed.size()-1 || flowerbed[i+1]==0)){
                flowerbed[i] =1;
                n--;
            }
            if(n==0){
                return true;
            }
        }
        return n<=0;
    }
};